accepted/2021/statics-in-interfaces/README.md

```diff
@@ -8,6 +8,793 @@ C# is looking at enabling static abstract members in interfaces (https://github.
+
+        // The majority of the IEEE required operations are listed below
+        // This doesn't include the recommended operations such as sin, cos, acos, etc
+
+        // 5.3.2
+        //  TSelf Quantize(TSelf x, TSelf y);
+        //  TSelf Quantum(TSelf x);
+
+        // 5.5.2
+        //  TDecEnc EncodeDecimal(TSelf x);
+        //  TSelf DecodeDecimal(TDecEnc x);
+        //  TBinEnc EncodeBinary(TSelf x);
+        //  TSelf DecodeBinary(TBinEnc x);
+
+        // 5.7.3
+        //  bool SameQuantum(TSelf x, TSelf y);
+    }
+}
+```
+
+### Interface Implementors
+
+Various types in the System namespace would implement the above interfaces as follows.
+
+```csharp
+namespace System
+{
+    public struct Byte : IUnsignedBinaryInteger<byte> { }
+
+    public struct Char : IUnsignedBinaryInteger<char> { }
+
+    public struct DateTime
+        : IAddable<DateTime, TimeSpan>,
+          IComparableOperators<DateTime, DateTime> // implies IEquatableOperators<DateTime, DateTime>
+    {
+        // MinValue and MaxValue should likely come from a base interface
+
+        static DateTime MinValue { get; }
+
+        static DateTime MaxValue { get; }
+    }
+
+    public struct DateTimeOffset
+        : IAddable<DateTimeOffset, DateTimeOffset>,
+          IAddable<DateTimeOffset, TimeSpan>,
+          IComparableOperators<DateTimeOffset, DateTimeOffset> // implies IEquatableOperators<DateTimeOffset, DateTimeOffset>
+    {
+        // MinValue and MaxValue should likely come from a base interface
+
+        static DateTimeOffset MinValue { get; }
+
+        static DateTimeOffset MaxValue { get; }
+
+        // TODO: DateTimeOffset defines an implicit conversion to DateTime, should that be modeled in the interfaces?
+    }
+
+    public struct Decimal
+        : ISignedNumber<decimal>
+    {
+        // Decimal defines a few additional operations like Ceiling, Floor, Round, and Truncate
+        // The rest of the IEEE operations are missing.
+
+        // Decimal and some other types likewise define "friendly" names for operators, should we expose them?
+    }
+
+    public struct Double : IBinaryFloatingPoint<double> { }
+
+    public struct Guid
+        : IComparableOperators<Guid, Guid> // implies IEquatableOperators<Guid, Guid>
+    {
+    }
+
+    public struct Half : IBinaryFloatingPoint<Half> { }
+
+    public struct Int16 : ISignedBinaryInteger<short> { }
+
+    public struct Int32 : ISignedBinaryInteger<int> { }
+
+    public struct Int64 : ISignedBinaryInteger<long> { }
+
+    public struct IntPtr : ISignedBinaryInteger<nint> { /* Size */ }
+
+    public struct SByte : ISignedBinaryInteger<sbyte> { }
+
+    public struct Single : IBinaryFloatingPoint<float> { }
+
+    public struct TimeSpan
+        : IAddable<TimeSpan, TimeSpan>,
+          IComparableOperators<TimeSpan, TimeSpan>, // implies IEquatableOperators<TimeSpan, TimeSpan>
+          IDivisable<TimeSpan, TimeSpan>,
```

Comment:
Doesn't this return a double, not a TimeSpan?

Reply:
Yes, I misread the docs.

Such an API would either require `IDivisable<TSelf, TOther, TResult>` or not being exposed via one of these interfaces. Exposing a `TResult` has been avoided so far to simplify things and make the APIs more usable in the most common case.

Reply:
I've changed this to have `IDivisable<TSelf, TOther, TResult>` as there are enough use cases in System that I think it justifies doing it from day 1